components/dashboards/org.wso2.carbon.dashboards.api/src/main/java/org/wso2/carbon/dashboards/api/internal/DashboardRestApi.java

```diff
@@ -139,8 +142,12 @@ public Response get(@PathParam("id") String id) {
     @Path("/")
     public Response create(DashboardMetadata dashboardMetadata) {
         try {
-            dashboardDataProvider.add(dashboardMetadata);
-            return Response.status(CREATED).build();
+            if (dashboardDataProvider.get(dashboardMetadata.getUrl()).equals(Optional.empty())) {
+                dashboardDataProvider.add(dashboardMetadata);
+                return Response.status(CREATED).build();
+            } else {
+                return Response.status(CONFLICT).build();
```

Comment:
As the 409 HTTP status code specifies, we need to send a message that describes "what is the conflict". See https://httpstatuses.com/409.
e.g. `Cannot create a new dashboard with URL ' + url + ' as  another dashboard exists for the same URL`

Reply:
This is addressed in the DashboardCreatePage.jsx
https://github.com/wso2/carbon-dashboards/pull/674/files#diff-b793316736f4eda10db2e68d97bd15a3R162

Reply:
My concern is regarding to REST API layer, not the UI layer. Regardless of what error message is shown in the UI, the REST API should return a proper error message with a error response, IMO.